src/main/scala/firrtl/transforms/ReplaceTruncatingArithmetic.scala

```diff
@@ -0,0 +1,53 @@
+package firrtl
+package transforms
+
+import firrtl.ir._
+import firrtl.Mappers._
+import firrtl.PrimOps._
+import firrtl.WrappedExpression._
+
+import scala.collection.mutable
+
+/** Replaces non-expanding arithmetic
+  *
+  * This replaces FIRRTL primops with ops that are not legal FIRRTL. They are useful for emission to
+  * languages that support non-expanding arithmetic (like Verilog)
+  */
+object ReplaceTruncatingArithmetic {
+
+  /** Mapping from references to the [[Expression]]s that drive them */
+  type Netlist = mutable.HashMap[WrappedExpression, Expression]
+
+  val SeqBIOne = Seq(BigInt(1))
+
+  def onExpr(netlist: Netlist)(expr: Expression): Expression = 
```

Comment:
Since these are public functions (that may be useful outside this transform), perhaps make the names more clear? Or maybe users should infer the behavior from its context, in that its in a `ReplaceTruncatingArithmetic` object. Food for thought. I do think its worthwhile to add scaladoc which says it looks at netlist, but doesn't update it.

Reply:
Yeah agreed. I'm not super keen on exposing functions that mutate their arguments (like `onStmt` does), but you previously (correctly) pointed out that they may be useful elsewhere.

I'll write Scaladoc

Reply:
Do you have any suggestions on better names for these? I'm wondering if the short names are okay by the virtue of the very specific enclosing object name `ReplaceTruncatingArithmetic`. In fact, I'd prefer to call them `apply` but then I'd have to use `mapExpr`, `mapStmt` etc. (although that might honestly be better).

Reply:
That is an interesting idea, I do think using mapExpr would clearer to users.